credentials/credentials-route.js

```diff
@@ -0,0 +1,233 @@
+const AWS = require('aws-sdk');
+var encryptor = require('file-encryptor');
+var fs= require('fs');
+const multer = require('multer');
+require("dotenv").config();
+const path = require('path');
+
+const s3 = new AWS.S3({
+  accessKeyId: process.env.REACT_APP_AWS_ACCESS_KEY_ID,
+  secretAccessKey: process.env.REACT_APP_AWS_SECRET_ACCESS_KEY,
+  region: process.env.REACT_APP_AWS_REGION
+});
+console.log("access id", process.env.REACT_APP_AWS_ACCESS_KEY_ID)
+module.exports = (app) => {
+
+const uploadDirectory = './uploads';
+if (!fs.existsSync(uploadDirectory)) {
+  fs.mkdirSync(uploadDirectory);
+}
+  // Configure Multer storage
+const storage = multer.diskStorage({
+  destination: (req, file, cb) => {
+    cb(null, 'uploads/');
+  },
+  filename: (req, file, cb) => {
+    const fileName = file.originalname.toLowerCase().split(' ').join('-');
+    cb(null, fileName);
+  }
+});
+
+// Create a Multer instance
+const upload = multer({ storage });
+
+app.post('/upload', upload.single('file'), (req, res) => {
+  
+    try {
+      const file = req.file;
+      const fileName = req.body.fileName;
```

Comment:
what if two users upload the same file? where is the metadata stored for mapping between files and users?

Reply:
since a unique key is used to store the files in the bucket and for encryption, we will not have any overrides here